pkg/datapath/linux/node.go

```diff
@@ -156,14 +156,20 @@ func deleteTunnelMapping(oldCIDR *cidr.CIDR, quietMode bool) {
 		return
 	}
 
-	log.WithField("allocCIDR", oldCIDR).Debug("Deleting tunnel map entry")
+	log.WithFields(logrus.Fields{
+		"allocCIDR": oldCIDR,
+		"quietMode": quietMode,
+	}).Debug("Deleting tunnel map entry")
 
-	if err := tunnel.TunnelMap.DeleteTunnelEndpoint(oldCIDR.IP); err != nil {
-		if !quietMode {
+	if !quietMode {
+		if err := tunnel.TunnelMap.DeleteTunnelEndpoint(oldCIDR.IP); err != nil {
 			log.WithError(err).WithFields(logrus.Fields{
 				"allocCIDR": oldCIDR,
 			}).Error("Unable to delete in tunnel endpoint map")
 		}
+	} else {
+		_ = tunnel.TunnelMap.SilentDeleteTunnelEndpoint(oldCIDR.IP)
```

Comment:
Should we be a bit extra defensive and only ignore the `ENOENT` error here?

Reply:
I think we won't ever see `ENOENT` or any other error here due to the following in `pkg/bpf.(*Map).deleteMapEntry`:

https://github.com/cilium/cilium/blob/67a3618d4775f0f52146a4c72693d0c2012f904a/pkg/bpf/map_linux.go#L977-L986

`SilentDeleteTunnelEndpoint` will call `TunnelMap.SilentDelete` which then calls `(*Map).deleteMapEntry` with `ignoreMissing = true`, meaning that we'll always get a `nil` error.